frontend/public/components/volumes-table.tsx

```diff
@@ -52,83 +39,107 @@ const ContainerLink: React.FC<ContainerLinkProps> = ({name, pod}) => <span class
-        <div className="col-lg-2 hidden-md hidden-sm hidden-xs">
-          { _.get(pod, 'kind') === 'Pod'
-            ? <ContainerLink name={m.container} pod={pod as PodKind} />
-            : <div>{m.container}</div>
-          }
-        </div>
-      </div>
-    </React.Fragment>)}
-  </div>;
+const VolumesTableHeader = () => {
+  return [
+    {
+      title: 'Name',
+      props: { className: volumeRoColumnClasses[0]},
+    },
+    {
+      title: 'Mount Path',
+      props: { className: volumeRoColumnClasses[1]},
+    },
+    {
+      title: 'SubPath',
+      props: { className: volumeRoColumnClasses[2]},
+    },
+    {
+      title: 'Type',
+      props: { className: volumeRoColumnClasses[3]},
+    },
+    {
+      title: 'Permissions',
+      props: { className: volumeRoColumnClasses[4]},
+    },
+    {
+      title: 'Utilized By',
+      props: { className: volumeRoColumnClasses[5]},
+    },
+    {
+      title: '',
+      props: { className: volumeRoColumnClasses[6]},
+    },
+  ];
 };
-VolumeRow.displayName = 'VolumeRow';
+VolumesTableHeader.displayName = 'VolumesTableHeader';
 
-export const VolumesTable: React.FC<VolumesTableProps> = ({podTemplate, heading}) => (
-  <React.Fragment>
-    {heading && <SectionHeading text={heading} />}
-    {_.isEmpty(podTemplate.spec.volumes)
+const VolumesTableRow = ({obj: volume, index, key, style}) => {
+  const type = _.get(getVolumeType(volume.volume), 'id', '');
+  const loc = getVolumeLocation(volume.volume);
+  const name = volume.name;
+  const permission = volume.readOnly ? 'Read-only' : 'Read/Write';
+  const { resource } = volume;
+  const pod = resource.kind === 'Pod'? resource : resource.spec.template;
+
+  return (
+    <TableRow id={index} index={index} trKey={key} style={style}>
+      <TableData className={volumeRoColumnClasses[0]}>{name}</TableData>
+      <TableData className={classNames(volumeRoColumnClasses[1], 'co-break-word')}>{volume.mountPath}</TableData>
+      <TableData className={volumeRoColumnClasses[2]}>{volume.subPath}</TableData>
+      <TableData className={volumeRoColumnClasses[3]}>
+        <VolumeIcon kind={type} />
+        <span className="co-break-word">{loc && ` (${loc})`}</span>
+      </TableData>
+      <TableData className={volumeRoColumnClasses[4]}>{permission}</TableData>
+      <TableData className={volumeRoColumnClasses[5]}>
+        {_.get(pod, 'kind') === 'Pod' ? <ContainerLink name={volume.container} pod={pod as PodKind} /> : <div>{volume.container}</div>}
+      </TableData>
+      <TableData className={volumeRoColumnClasses[6]}>
+        <VolumeKebab actions={menuActions} kind={resource.kind} resource={resource} rowVolumeData={volume} />
+      </TableData>
+    </TableRow>
+  );
+};
+VolumesTableRow.displayName = 'VolumesTableRow';
+
+export const VolumesTable = props => {
+  const { resource } = props;
+  const data = getRowVolumeData(resource);
+  const pod = resource.kind === 'Pod' ? resource : resource.spec.template;
```

Comment:
We have this logic in many places. Can we create a helper?

Reply:
Created a method to reduce identical logic is so many places.